core-object.js

```diff
@@ -19,20 +22,68 @@ CoreObject.extend = function(options) {
 
   function Class() {
     var length = arguments.length;
+    var instanceMixin = this.__instanceMixin__;
+
+    for (var key in instanceMixin) {
+      this[key] = instanceMixin[key];
+    }
+
+    Class.wasApplied = true;
```

Comment:
i would love to error, if someone tries to reopen after `wasApplied` becomes true